homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,717 @@
+    translation_key: str = field(default=CoverDeviceClass.BLIND.value, init=False)
+    device_class: CoverDeviceClass = field(default=CoverDeviceClass.SHADE, init=True)
+
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.ROLLER.value: MotionCoverEntityDescription(),
+    MotionBlindType.HONEYCOMB.value: MotionCoverEntityDescription(),
+    MotionBlindType.ROMAN.value: MotionCoverEntityDescription(),
+    MotionBlindType.VENETIAN.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.VENETIAN_TILT_ONLY.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.BLIND
+    ),
+    MotionBlindType.DOUBLE_ROLLER.value: MotionCoverEntityDescription(),
+    MotionBlindType.CURTAIN.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN
+    ),
+    MotionBlindType.VERTICAL.value: MotionCoverEntityDescription(
+        device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up blind based on a config entry."""
+
+    blind_type = BLIND_TO_ENTITY_TYPE[entry.data[CONF_BLIND_TYPE]]
+    blind = blind_type(entry)
+
+    hass.data[DOMAIN][entry.entry_id] = blind
+
+    async_add_entities([blind])
+
+
+class GenericBlind(CoverEntity):
+    """Representation of a blind."""
+
+    device_address: str
+    device_rssi: int | None = None
+    _device: MotionDevice
+    _attr_connection_type: MotionConnectionType = MotionConnectionType.DISCONNECTED
+    _running_type: MotionRunningType | None = None
+
+    _use_status_position_update_ui: bool = False
+
+    _battery_callback: Callable[[int | None], None] | None = None
+    _speed_callback: Callable[[MotionSpeedLevel | None], None] | None = None
+    _connection_callback: Callable[[MotionConnectionType | None], None] | None = None
+    _signal_strength_callback: Callable[[int | None], None] | None = None
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize the blind."""
+        _LOGGER.debug(
+            "(%s) Setting up %s cover entity (%s)",
+            entry.data[CONF_MAC_CODE],
+            entry.data[CONF_BLIND_TYPE],
+            BLIND_TO_ENTITY_TYPE[entry.data[CONF_BLIND_TYPE]].__name__,
+        )
+        super().__init__()
+        self.entity_description = COVER_TYPES[entry.data[CONF_BLIND_TYPE]]
+        self.config_entry: ConfigEntry = entry
+        self.device_address: str = entry.data[CONF_ADDRESS]
+        self._attr_name: str = ENTITY_NAME.format(mac_code=entry.data[CONF_MAC_CODE])
+        self._attr_unique_id: str = entry.data[CONF_ADDRESS]
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            identifiers={(DOMAIN, entry.data[CONF_MAC_CODE])},
+            manufacturer=MANUFACTURER,
+            name=self._attr_name,
```

Comment:
Is it not possible to determine a `model` for the device?

Reply:
No it is not, the MotionBlinds BLE app also asks the user to select what kind of device they have.

Reply:
Okay, but at least the blind type is already known when we set the device data. Does a specific `BlindType` correspond to a single device model and if not how about setting the model name to something like "MotionBlinds <BlindType> Device"? Just an idea.

Reply:
Right, I could set it to the `BlindType`, I think that makes sense

Reply:
Do you know if there is a way to provide a translation in strings.json for the model? Otherwise I'll just adjust the enum values

Reply:
I looked at the localization docs and I didn't find anything about the device model unfortunately, so it's probably not possible.

Reply:
In that case I'll just change the values of the `MotionBlindType` enum to better display model names